experiment_prototype/scan_classes/sequences.py

```diff
@@ -296,8 +295,23 @@ def __init__(self, seqn_keys, sequence_slice_dict, sequence_interface, options):
         self.seqtime = 2*self.options.tr_window_time + self.pulses[-1]['pulse_timing_us'] + \
                        self.last_pulse_len
 
-        # TODO change numberofreceivesamples to oversample so that no sample dropping
-        # has to happen due to edge effects (account for filter length).
+        blanks = []
+        sample_time = 1.0/float(self.options.output_sample_rate)
+        first_sample_time = sample_time/2.0  # TODO this needs to be calculated appropriately
+        # given number of samples added on the front when filtering occurs
+        pulses_time = []
+        for pulse in self.pulses:
+            pulse_start_stop = [pulse['pulse_timing_us'], pulse['pulse_timing_us'] + pulse[
+                'pulse_len']]
+            pulses_time.append(pulse_start_stop)
+        output_samples_in_sequence = int(self.seqtime/sample_time)
+        sample_times = [first_sample_time + i*sample_time for i in range(0,output_samples_in_sequence)]
+        for sample_num, time_s in enumerate(sample_times):
+            for pulse_start_stop in pulses_time:
+                if time_s in range(pulse_start_stop[0], pulse_start_stop[1]):
```

Comment:
Should time_s be in this range or should it just be time_s >= pulse_start_stop[0] and time_s <= pulse_start_stop[1]?